lib/HAL/Compile/ThreadedCompilationManager.cpp

```diff
@@ -242,6 +245,7 @@ llvm::Error ThreadedCompilationManager::compileMLIR(mlir::ModuleOp moduleOp) {
   auto err = walkTargetModulesThreaded(&target, moduleOp, targetsTiming,
                                        threadedCompileMLIRTarget,
                                        postChildrenEmitToPayload);
+  emitTargetDiagnostics();
```

Comment:
Instead of emitting here we would just collect the diagnostics at each stage in the walk.

Reply:
Updated in https://github.com/openqasm/qe-compiler/pull/240/commits/07c5019bb53affc143bef6f8ff87faf93c42426d